CODINGSTYLE.md

```diff
@@ -196,6 +196,90 @@ In the following example, the empty string is the sentinel value indicating fail
 std::string parse_bss_info(const std::string &bss_info_str, db::bss_info_conf_t &bss_info_conf);
 ```
 
+Prefer to use multiple points of exit from functions.
+One might argue that it is less confusing if there is only a single exit point, but it leads to more nesting and therefore more indentation which makes the code less readable.
+Another advantage of using multiple points of exit is that you can check preconditions and exit early at the start of a function, so that you know in the body of the function that certain conditions are true, without the entire body of the function being indented 5 miles off to the right. This usually minimizes the number of scopes you have to worry about, which makes code much easier to follow.
+Multiple points of exit means that you can exit anywhere you please. This used to be more confusing in the old days, but now that we have syntax-coloring editors and compilers that detect unreachable code, it's a lot easier to deal with.
+
+Avoid using `goto` - The goto failure pattern is a corollary of the multiple exit points. It applies when there is cleanup to do. But since we use C++, there should never be any cleanup to do (it should all be handled by automatic variables).
+
+Examples:
+
+Good - multiple points of exit:
+
+```cpp
+bool func()
+{
+    if (something) {
+        // do something
+        return true;
+    } else if (something else) {
+        // do something else
+        return false;
+    } else {
+        return false;
+    }
```

Comment:
IMO this doesn't really show your point of keeping the indentation level low or checking arguments early, so maybe we should keep only the other example.
On the other hand, it shows what having multiple points of exit _means_, I guess that's why you did it like this.